doc/contribute/guidelines.rst

```diff
@@ -536,12 +536,28 @@ reference manuals, etc.
 Coding Style
 ============
 
+.. note::
+   Coding style is enforced on any new or modified code, but contributors are
```

Comment:
I think this is wrong for code modifications. When code is modified the resulting final code will not comply with the coding style, so preference should be given to consistency. Reviewers should in this case not only review the changes but also how this new code integrates with the existing code base. In many cases it will be possible to align the new code with the coding style, but not always. A gentle nudge in the direction of improved coding style when possible should be sufficient.

Reply:
"This is not a legally binding contract" wait what? wonder what I signed then :D (joke)

I'm fine with the wording as is as well

Reply:
If it is intended to be just a guideline, what about reformulating it as:

Coding style is enforced on new additions (drivers, subsystems, ...). For modifications to the existing code base contributors are encouraged to align modifications with the coding style.

Reply:
Disagree, we've been over in a previous comment in this thread, things should not be being added that are wrong just because there is previous code that is wrong, you add more bad code, more technical debt and burden someone else with having to fix it